pkg/ibm/vpc_collection.go

```diff
@@ -45,7 +45,7 @@ func iteratePagedAPI[T any, Q HasGetNextStart[T]](
 }
 
 //nolint:dupl // The duplication is essentially creating the adapter
-func getVPCs(vpcService *vpcv1.VpcV1, region string) ([]*datamodel.VPC, error) {
+func getVPCs(vpcService *vpcv1.VpcV1, region, resourceGroupID string) ([]*datamodel.VPC, error) {
 	APIFunc := func(pageSize int64, next *string) (*vpcv1.VPCCollection, any, error) {
 		return vpcService.ListVpcs(&vpcv1.ListVpcsOptions{Limit: &pageSize, Start: next})
```

Comment:
```suggestion
		return vpcService.ListVpcs(&vpcv1.ListVpcsOptions{Limit: &pageSize, Start: next, ResourceGroupId: resourceGroupId})
```
Doesn't it work this way?

Reply:
yes it works, I will do the for all of the relevant resources

Reply:
but still need to check if the arg is empty